decidim-core/app/jobs/decidim/machine_translation_create_fields_job.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Decidim
+  class MachineTranslationCreateFieldsJob < ApplicationJob
+    queue_as :default
+
+    def perform(id, resource_type, field_name, field_value, locale)
+      Decidim::TranslatedField.create!(
+        translated_resource_id: id,
+        translated_resource_type: resource_type,
+        field_name: field_name,
+        field_value: field_value,
+        translation_value: nil,
+        translation_locale: locale
+      )
+      Decidim::DummyTranslator.translate(id, locale, field_name, field_value)
```

Comment:
Why are we hardcoding the `DummyTranslator`?
We need to change this to some factory that returns the configured external system no?

Reply:
Agree, I'd do it similar to the geocoding service or the sms gateway

Reply:
So, what we need to do is configure the translation service from the Decidim initializer (this means updating the app generator and upgrading notes for existing applications).

Something like:

```ruby

Decidim.configure do |config|
  config.machine_translation_service = Decidim::DummyTranslator
end
```

And from the job:

```ruby
def perform(...)

  return unless Decidim.machine_translation_service.present?

  ## continue normally
end
```

Check out https://github.com/decidim/decidim/blob/master/decidim-generators/lib/decidim/generators/app_templates/initializer.rb#L65 as en example, and this PR:

https://github.com/decidim/decidim/pull/4429